### ISSUE TITLE
linter: consistent-test-it considers sibling `describe` as parent

### ISSUE THREAD

**Issue by ipanasenko** [oxc-project/oxc#22633]
### What version of Oxlint are you using?

1.63.0

### What command did you run?

nothing special, default settings

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```js
module.exports = {
  plugins: ['jest', 'vitest'],
  rules: {
    'jest/consistent-test-it': ['error', {fn: 'test', withinDescribe: 'it'}],
    'vitest/consistent-test-it': ['error', {fn: 'test', withinDescribe: 'it'}],
  },
};
```

### What happened?

Standalone `test` is reported as if it's inside `describe`

<img width="834" height="372" alt="Image" src="https://github.com/user-attachments/assets/0bb98dc5-7dd4-4c0a-b481-f69d1f380176" />

Potential fix, but I couldn't verify it on my laptop: https://github.com/oxc-project/oxc/pull/22632
